src/wasmv1_execution/abi/abis.rs

```diff
@@ -190,17 +191,20 @@ pub(crate) fn abi_call(
             let address = req.target_sc_address.ok_or_else(|| {
                 WasmV1Error::RuntimeError("No address provided".into())
             })?;
+            let address = native_address_to_string(handler, address)?;
```

Comment:
converting to string allows to reuse the call_coins function as is (and other functions).
it implies that we convert to string every time.
does it looks ok for you?

Reply:
It's a great fit for now but I'm not sure it should always be like that

Reply:
Yes same feeling
My goal is to have the abis done (so the interface to the SC world is ok)
Then we'll be able to refactor the internal parts

Reply:
I agree